### PR TITLE
RES: ignore atom use items on 2018 edition

### LINKED ISSUE #3846: the parking_lot crate does not receive syntax highlighting nor display of object type
For some reason, the lock() function of a parking_lot::Mutex<T> does not display. It exists, but the IDE does not pick it up

### PR DESCRIPTION
**Pull request by vlad20012**
Fixes #3846

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/resolve/ItemResolution.kt [hunk 1]
```diff
@@ -99,14 +99,20 @@ fun processItemDeclarations(
         }
     }
 
+    val isEdition2018 = scope.isEdition2018
     for (speck in itemImports) {
         check(speck.useGroup == null)
         val path = speck.path ?: continue
         val name = speck.nameInScope ?: continue
+        if (isEdition2018 && speck.qualifier == null && path.referenceName == name && path.path == null &&
```
[the remark this answers is not in the document]
Comment (Undin) [reply src/main/kotlin/org/rust/lang/core/resolve/ItemResolution.kt]: Let's add a comment why we need this

File: src/test/kotlin/org/rust/lang/core/resolve/RsPackageLibraryResolveTest.kt [hunk 2]
```diff
@@ -566,4 +566,15 @@ class RsPackageLibraryResolveTest : RsResolveTestBase() {
...
+    // Issue https://github.com/intellij-rust/intellij-rust/issues/3846
+    @MockEdition(CargoWorkspace.Edition.EDITION_2018)
+    fun `test extra use of crate name`() = stubOnlyResolve("""
+    //- dep-lib/lib.rs
+        pub struct Foo;
+    //- lib.rs
+        use dep_lib_target;
+        use dep_lib_target::Foo;
+                          //^ dep-lib/lib.rs
+    """)
```
Comment (Undin) [inline src/test/kotlin/org/rust/lang/core/resolve/RsPackageLibraryResolveTest.kt]:
```suggestion
    """, ItemResolutionTestmarks.extraAtomUse)
```

**Comment by vlad20012** [comment]
I found that it fails on `use foo::{self}` case...

File: src/main/kotlin/org/rust/lang/core/resolve/ItemResolution.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (vlad20012) [reply src/main/kotlin/org/rust/lang/core/resolve/ItemResolution.kt]: `speck.qualifier` and `path.path` are different things:
```rust
use foooooooooo::{baaaaaaaar::baz}
//  ^speckq        ^path.path ^path
```

**Comment by vlad20012** [comment]
Fixed. I also covered `use foo::{self}` case

File: src/main/kotlin/org/rust/lang/core/resolve/ItemResolution.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Undin) [reply src/main/kotlin/org/rust/lang/core/resolve/ItemResolution.kt]: Indeed!
I've mixed it up with `RsPath.qualifier`

**Pull request merged**